File: 2nd_quarter/week05/Royce/모의 고사.py
def solution(answers):
    answer = []
    
    arr1 = [1,2,3,4,5]
    arr2 = [2,1,2,3,2,4,2,5]
    arr3 = [3,3,1,1,2,2,4,4,5,5]
    
    num1 = 0
    num2 = 0
    num3 = 0
    for i in range(len(answers)):
        if answers[i] == arr1[i % len(arr1)]:
            num1+=1
        if answers[i] == arr2[i % len(arr2)]:
            num2+=1
        if answers[i] == arr3[i % len(arr3)]:
            num3+=1
    num = [num1, num2, num3]
    
    for idx, s in enumerate(num):
        if s == max(num):           ## 최대 값과 같을 경우
            answer.append(idx+1)    ## 그 경우의 인덱스를 넣어준다.
    
    return answer

File: 2nd_quarter/week05/whaeun/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int arr1 [5] = {1, 2, 3, 4, 5};
int arr2 [8] = {2, 1, 2, 3, 2, 4, 2, 5};
int arr3 [10] = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    
    int score1 = 0, score2 = 0, score3 = 0;
    
    for(int i = 0; i < answers.size(); i++){
        if(arr1[i % 5] == answers[i]){
            score1++;
        }
    }
    
    for(int i = 0; i < answers.size(); i++){
        if(arr2[i % 8] == answers[i]){
            score2++;
        }
    }
    
    for(int i = 0; i < answers.size(); i++){
        if(arr3[i % 10] == answers[i]){
            score3++;
        }
    }
    
    vector<pair<int, int>> v;
    
    v.push_back({score1, 1});
    v.push_back({score2, 2});
    v.push_back({score3, 3});
    
    sort(v.begin(), v.end());
    reverse(v.begin(), v.end());
    
    if(v[0].first > v[1].first){
        answer.push_back(v[0].second);
    }
    else if (v[0].first == v[1].first && v[1].first == v[2].first){
        answer.push_back(v[0].second);
        answer.push_back(v[1].second);
        answer.push_back(v[2].second);
    }
    else{
        answer.push_back(v[0].second);
        answer.push_back(v[1].second);
    }
    
    sort(answer.begin(), answer.end());
    
    return answer;
}
